### ISSUE TITLE
[Notice] "Hide DS4 Controller" option Removed

### ISSUE THREAD

**Issue by Ryochan7** [Ryochan7/DS4Windows#1520]
I have been thinking about it for several months but a query finally made me decided to remove the old "Hide DS4 Controller" option from DS4Windows. The only semi-supported method of obtaining exclusive access to a controller in future releases will be through HidGuardian.

https://github.com/Ryochan7/DS4Windows/commit/783102f5b8a1761532f482f5c4d56dde2f14508a

**Comment by CarissaIsWierd** [comment]
Why though? I've been using DS4Windows with 'Hide DS4 Controller' always ticked for 3 years. It's easy and makes sense.

I looked at the setup guide you have for HidGuardian (https://github.com/Ryochan7/DS4Windows/wiki/DS4Windows-and-HIDGuardian-Install-and-Setup-Guide) and it seems a lot more complicated. Is it even up-to-date? There's mention of a helper app, but it seems the same as HidNinja in DS4Windows. It also says to select 'Write HID Guardian Device List' under the Settings tab, but that doesn't exist.

After using the HidGuardian Test Installer, rebooting, and unticking 'Hide DS4 Controller', none of my games that use an Auto Profile with Emulated Controller set to DS4 work anymore. I must be doing something wrong, or it's just not as intuitive as 'Hide DS4 Controller' is.

**Comment by mika-n** [comment]
@CarissaIsWierd   Does things work if you use a profile with x360 output controller type?
Do you happen to have DS4 revision1 (DS4Windows logfile shows DS4v1 gamepad type)?
In that case take a look at the following post. It exlains a scenario where HidGuardian hides both the physical controller AND the output virtual gamepad controller (leaving games without any gamepad). The solution is to tweak HidG settings to hide only DS4v2 gamepad. 
https://github.com/Ryochan7/DS4Windows/issues/669#issuecomment-487302990

But, if you have DS4v1 gamepad then you have a problem with HidGuardian when output virtual controller is DS4 and not xbox360.

**Comment by CarissaIsWierd** [comment]
@mika-n Yeah, mine are official DS4v1. They do seem to work fine with HidGuardian if I'm not trying to use the option to emulate them as DS4. I use that for every new game that allows it, which most seem to. When I first started using DS4Windows, there were a lot of games getting double button presses, making me keep the option always ticked. Though I can't remember which of them did so to be able to test again, if anything has changed..

Would it be so bad to just keep the 'Hide DS4 Controller' option? ༼ つ ◕_◕ ༽つ

**Comment by Sakitoshi** [comment]
@CarissaIsWierd you want to use hidninja to allow only ds4windows and steam to see your controller.
when you want x360 emulation, you start ds4windows and when you want to use steam, you stop ds4windows.
that's exactly how I'm using it and I have no problems snagging the controller from steam just pressing start on ds4windows.
but for this to work, the "hide ds4 controller" option needs to be ticked or else steam will see the ds4 and the emulated x360 controller.
so I agree that the option needs to stay or made permanent since I can't think of a scenario where you want to have a duplicate controller, it only causes trouble.

**Comment by mika-n** [comment]
The question is not about if the physical gamepad device handle needs to be hidden or not. The question is if this hiding should be done via the old "Hide DS4 Controller" option or via HidGuardian device driver.
https://github.com/Ryochan7/DS4Windows/wiki/Exclusive-Mode-(Hide-DS4-Controller-config-option)-tips-and-issues

"Hide DS4 Controller" option is optional way if HidG driver is not used to do the same. The problem with HideDS4Controller (=exclusive mode) is that nowadays it often fails because of issues beyond the control of DS4Windows app (as explained in the above linked web page. DS4 hijacking apps come and go). HidG is more fool proof way to do the hiding thing.

 However, owners of DS4v1 gamepad have an issue with profiles using DS4 output mode because HidG (the current version) cannot tell the difference between virtual output ds4 gamepad device and the ds4v1 physical gamepad. So, HidG hides both of these. Owners of DS4v2 gamepad don't have this issue with HidG.

**Comment by Sakitoshi** [comment]
I do own two v1 controllers (launch ps4 and a blue one bought as soon as colors were available) so I can do some tests.

from what I can see, since the real and emulated ds4 both have exactly the same hardware identifiers, there is no way for hidguardian to tell which is the emulated controller.
the only solution I can think off is to emulate a v2 controller for v1 hardware and emulate a v1 controller for v2 hardware, but of course, I'm no expert.

as for what I said in the previous post.
I use hidguardian and "hide ds4 controller" in conjunction to let steam see only the emulated ds4 controller.
I whitelist steam so he can see the controller, then check "hide ds4 controller" so ds4windows can hide the real controller from steam. this is needed because of how things are working right now.
though I'm currently leaving ds4windows disabled when I use my ds4 with steam since the emulated ds4 has the lightbar and rumble disabled. I still need this setup to let steam see the real controller.

**Comment by escape75** [comment]
I've always used "Hide DS4 Controller" option under Settings with my DS4 v2,
and I use that for all my games, it's never failed me, so why remove something
that a lot of people find beneficial ? Maybe add a warning while clicking on it ?

**Comment by Skynet19972029** [comment]
Same here. Been using Exclusive mode since day one I had the controller and never had an issue :(

**Comment by Yohoki** [comment]
I just finally got exclusive mode working correctly every time! Nooooo! I literally just uninstalled geforce experience 2 days ago. xD

Kidding aside....
This isn't in the current build, 2.1.15, correct? I've got 4 controllers and only 1 of them is a v2 controller... so I'm going to have to stay on the highest version with exclusive mode. It would be fine if I only used my v2 controller, but I've got 3 kids and we all sit down and play some of these games together. Exclusive mode was the only thing making some of these games behave correctly. Such as Diner Bros, which accepts ds4 input, but needs remapped buttons (X and O turn into [ ] and X without remapping.). I'm sure there are a plethora of games that this will effect.

It's unfortunate that it's such a troublesome feature. It really does work well when it does work. The hidguardian solution would be fine if it were able to distinguish the DS4v1 from virtual. I would much rather see the feature still included, but be made aware that the feature is depreciated, no longer supported, and it may or may not work.

**Comment by Xseba360** [comment]
I also have been using this feature since day one. Some games_""support""_ both directinput and xinput which tends to be troublesome really and it all depends which one is detected first by the game. Please do bring this option back :(

**Comment by Darthagnon** [comment]
Please don't remove it :'( I use this option, too.

**Comment by Ryusennin** [comment]
I don't quite understand the resistance.

"Hide DS4" is an old hack that has made more harm than good. Hundreds of people have complained on this forum that this old exclusive mode didn't work for most modern games.

HidGuardian completely replaces this hack and works flawlessly in all situations.

**Comment by Skynet19972029** [comment]
The method in the guide for Hidguardian it's too convoluted and the last time I tried to do it, it just didn't work.
Maybe it needs an update?
I don't know.

**Comment by Yohoki** [comment]
My resistance is that I use DS4 v1 controllers. Unless  HIDGuardian can hide the actual DS4 without affecting the Virtual DS4 , then it is not "Flawless in all situations". We don't all have the money to upgrade all of our controllers.

Yes, it's a pain in the butt. It's hard to set up. That is not the fault of DS4Win, though. That is other programs interfering. You can't control what programs other people are using. If set up properly, "Hide DS4" works 100% of the time, completely flawlessly, regardless of hardware, and with a simple click of a button. You cannot say the same about HIDGuardian.

I agree, that the "Hide DS4" option causes issues because it's not easy to tell what programs are stealing exclusive access for absolutely no reason. But again, that is not something caused by DS4W. That is being caused by other programs that are NOT functioning correctly. Why does Edge of GeForce need exclusive access to the controller? Not for any reason I can think of.

I'm using older controller mixed with a v2 controller. So even HIDGuardian will not work for me. I'm going to have to stick to this version of DS4 and just not upgrade... because upgrading will now break a lot of my setups on multiple devices now...

**Comment by mika-n** [comment]
The author of HidGuardian has a new version of HidGuardian in a lab, so hopefully it resolves some of those issues with the current HidGuardian version. Even the original author of HidG (Nefarious) doesn't recommend to use this existing (old) version of HidG unless user is ready to take the risk and understands how it works. Anyway, the existing version of HidG is the best thing we have. 

Ever wondered why there are no other similar solutions like HidGuardian? What it does is veeeeeeeeeery complex thing to do, because of kernel driver API, blocking access to a device handle, etc. Not many people even understand what it does under the hood in WinOS kernel level.

"Hide DS4 Controller" (=exclusive access to device handle in user-space app level, not in kernel API level) works well when it works, but many people don't even bother to read the warning message and the linked "exclusive access" Wiki doc page when it doesn't work. Instead they just send a private message or post yet-another issue ticket asking the same thing discussed several times already. Maybe there should be a better warning/disclaimer msg in DS4Windows app when the exclusive access fails to fight this documentation issue.

**Comment by Yohoki** [comment]
Since I depend on Exclusive Mode, I've created a fork to re-enable this option on my own github. It's currently updated to the most current commit. I'll probably update it every time there's a new release, since commits are frequent and sometimes get rolled back. So, for those of us that still rely on Exclusive mode, you're welcome.

The only change I've made to the original code is to add a message box when enabling Exclusive mode, warning that this is an outdated feature and is no longer supported. If people can't get it working, I offer no support. So google the issue yourself if it's causing issues, because I've no idea how to fix it for you, and really don't care to help either.

https://github.com/Yohoki/DS4WindowsEx

**Comment by RaydenX93** [comment]
What about making "Hide DS4 Controller" option hidden by default and making it appear only when DS4Windows is launched with a particular command-line argument?

This way, only users that ACTUALLY want the feature can use it and new users will probably never know about it.

**Comment by mika-n** [comment]
@Ryochan7   Hmmmm.... Maybe "Hide DS4 Controller" option should be kept here until the new version of HidGuardian is released?  And maybe doing the proposed "hidden option", so only users who actually bother to read documentation and warnings about the exclusive mode find out and are able to enable that option?

The new HidG presumably solves many of the issues in the current HidGuardian architecture (for example the virtual DS4 rev1 is not hidden by the new HidGuardian version as it happens with the current HidG version, controllers are hidden per controller instance and not by controller type).

**Comment by Yohoki** [comment]
My branch has been merged into the main branch now as https://github.com/Ryochan7/DS4Windows/commit/354b58ff6b8d5a113dde4db9ae0776ab992897b1. So, the "Hide DS4 Controller" option has been re-enabled, but there is now a warning box that will pop up when enabling it. It states "This feature is depreciated and no longer supported. Exclusive mode usage is provided merely as a legacy feature. Do NOT ask for help for this feature, you will not receive any."

**Comment by Ceo22** [comment]
> @CarissaIsWierd you want to use hidninja to allow only ds4windows and steam to see your controller.
> when you want x360 emulation, you start ds4windows and when you want to use steam, you stop ds4windows.
> that's exactly how I'm using it and I have no problems snagging the controller from steam just pressing start on ds4windows.
> but for this to work, the "hide ds4 controller" option needs to be ticked or else steam will see the ds4 and the emulated x360 controller.
> so I agree that the option needs to stay or made permanent since I can't think of a scenario where you want to have a duplicate controller, it only causes trouble.

Could you please explain to me how to use HidNinja ?. I didn't really find any information and I'm interested in how I can make only DS4Windows and Steam have access to the controller.

**Comment by gigabit942007** [comment]
Regarding the importance of this feature.
Due to differences in layout DS4 vs XBOX360 controller if i try to play Sekiro Shadows Die Twice with this option unchecked it result in the camera rotating continuosly. Could this option be implemented per profile or maybe another solution found? Sekiro does not recognise the DS4 even though it sees the DS4 axis for the triggers.

**Comment by Ryusennin** [comment]
You should open a specific issue, this a generic discussion.

Also: https://github.com/Ryochan7/DS4Windows/issues?q=sekiro

**Comment by RyuMakkuro** [comment]
I've seen some very alarming thoughts posted here about this and decided that it warranted a post because the direction you're taking with this would effectively force me to move to Steam controller support, which I would absolutely hate to do so.

So let's start with the fact that multi input device support is a feature found most commonly in car games with wheel support and games with local multiplayer, aka split screen or hot seat. In racing games when you use a wheel, you also have pedals then as an extra gear stick, maybe handbrake as well and quite a lot of extra small things are possible to add as well. All of them are different devices and therefore require said games to support multiple inputs at the same time. So them picking up multiple inputs is not a bug, it's a very intended and necessary feature for said games to function properly. However said feature effectively makes it impossible to use any emulated controller in them since you'll have double input all the time, thus a solution that provides "exclusive" access is required. It's not an optional thing, it's absolutely required for normal operations. Also it's extremely important to realise that this isn't the fault of games but the inherent fault of the emulation software.



Second thing would be a very simple example from a DS4v1 user viewpoint:
Hide DS4 - works
HIDGuardian - doesn't work

I'm not exactly sure if there are some developer circles where replacing a feature that works with one that doesn't is considered an upgrade or improvement of any sort whatsoever. But I imagine any self respecting developer would absolutely reject said concept outright.
To give an example of what you're doing here with trying to force HIDGuardian which effectively removes DS4v1 support, it's like nVidia would suddenly release new drivers that would make all the GPUs that aren't the latest series completely non functional. It's effectively the "I can't be arsed anymore" statement. If you don't want to deal with constant reports about it not working, then include a warning or some very obvious notification that informs users that if they have any problems with this feature, they have to figure it out on their own what is causing issues as it is case by case situation. You can also point them where to start with like Shadowplay, Origin, Steam, etc. and the good ol' restarting the PC if all else fails.
You haven't tried anything at all to limit the amount of reports you get about this "not working", yet you already go straight into removing the absolutely mandatory feature altogether. Reminds me of situations where game developers get complaints from players about a feature not working correctly and their response to that is to remove it entirely, negatively affecting the entire game in the process.



Last thing is a bit different but depending on the view point, it may be more important than the previous two as those were about the software working at all, whereas this one is about users being able to use it at all. Yes, user friendliness.
Ignoring the fact HIDGuardian doesn't work, it is absolutely not a feature that is user friendly in any way, shape or form. You need to have software engineering degree to understand what's what, let alone to make it do what you want to do. If you've made this software for other developers, then that is perfectly fine. But considering the original DS4Windows idea was to be simple and easy tool to make your DS4 work in games, then HIDGuardian is the exact opposite of it. In fact I'd say that anything that requires more than 3 clicks that aren't as simple as "Add" -> "Choose controller" -> "Pick game" is absolutely unacceptable for this specific case. If I am forced to read through documentation or watch a video tutorial to make a gamepad work in a game, then I am absolutely not gonna bother and go for a solution I know that doesn't require it, despite being inferior in every other way. Just because it is simple and it works.
Of course this applies only to getting it to work, as I am perfectly happy with having plethora of options to adjust everything to suit my specific needs which is why I am using DS4Windows and hoping to use it for a very long time still. In that regard everything about it is as good as it gets... well with the exception that you confused sensitivity with saturation and effectively have two features doing the same thing (both max zone and sensitivity are saturation) where nothing actually affects sensitivity (sensitivity is the rate at which it follows player input, aka how quickly, not how much).
Now to put this into a better perspective, I do have some IT knowledge and I don't mind doing some tinkering to get something to work. That's me however and not what the vast majority of people are. Said majority that has issues reading anything longer than few words, let alone understanding it. So your current approach is a big red flag for a lot of people. And let's be honest, this software should be as plug and play as possible. All we want is our gamepad to work in games without issues.

I hope you will re-evaluate your stance on removing this feature and how you want to go about replacing it, because I genuinely don't want to use that blasted Steam support.

**Comment by Ryusennin** [comment]
HidGuardian _does_ work with DS4 v1. You just don't get support for "emulated DS4" and DS4 prompts.

And maybe you didn't read the 2.1.16 release notes, but the old Hide DS4 Controller option has been re-enabled (contribution by Yohoki some posts above yours). A warning message has been implemented to indicate that this feature is deprecated.

**Comment by RyuMakkuro** [comment]
> HidGuardian _does_ work with DS4 v1. You just don't get support for "emulated DS4" and DS4 prompts.

To me that is the same as doesn't work then. In situations like these, I'm either it works fully, or not at all.

> And maybe you didn't read the 2.1.16 release notes, but the old Hide DS4 Controller option has been re-enabled (contribution by Yohoki some posts above yours). A warning message has been implemented to indicate that this feature is deprecated.

I do read the changelog with new updates. I'm not talking what you did right now, I'm talking about the thought process you had for even thinking about removing said option and what steps had been done before you decided to do so (aka none) and your thought process for future. It would appear that maybe it was you who didn't read my post, albeit not that surprising given the wall of text it is.

**Comment by mika-n** [comment]
HidGuardian is more reliable way to hide the original physical controller, but the current version of it has issues with DS4v1 gamepads when a profile uses dualshock4 mode. DS4v2 gamepads work OK even in ds4 mode.

The problem is that the old HideDS4Controller option generates tons of messages from users who don't bother to read instructions and documentation. These days there are more and more apps and background services in WinOS hijacking the gamepad devices and preventing HideDS4Controller (=exclusive mode) to do its magic. Then users bomb developers with the same messages over and over again. Maybe you could help here and keep eye on those messages and explain those people what they need to do and make them to understand that if HideDS4Controller doesn't work then it is not a bug in DS4Windows but a "feature" of their PC setup?
https://github.com/Ryochan7/DS4Windows/wiki/Exclusive-Mode-(Hide-DS4-Controller-config-option)-tips-and-issues

After all DS4Windows is provided free of charge and people develop it on their own freetime as volunteers. If you don't like what the app does then feel free to program your own app instead of ranting. Or provide help to improve documentation and helping people on these forums to understand why HideDS4Controller option doesn't work always and releasing the time of developers to do the actual coding instead of answering the same question 100+ times.

**Comment by Yohoki** [comment]
To be fair.... It's his project. He can add or remove any parts to hit that he sees fit. If he thinks exclusive mode is a hassle, he can remove it. And let's be real, it IS a pain in the butt to get working and seemingly unrelated things break it. It's a hassle to have something in your program that doesn't work good... Not only does it mean people will keep asking questions, but it just doesn't feel good to know your code is broken and you don't know a way to fix it.

I really didn't expect @Ryochan7 to add it back in (my pull request message just said "Meh, it's worth a try ¯\_(ツ)_/¯"). I fully expected to have a second DS4Win program that I'd have to constantly update to re-add Exclusive mode for the few people that rely on it. But, if any newer updates he made had broken Exclusive mode, I would have had a lot of trouble adding it back in. So, hopefully the new (very annoying and in your way) dialogue box warning message will deter some people from messaging about the feature. (obviously, it didn't work THAT well as someone's already asked about Sekiro XD )

Hopefully, soon they'll release the new version of HIDGuardian that sounds like it'll fix the issue with v1 controllers, so that the users with mixed versions aren't left out. Then we can make a good tutorial for how to set it up, because I sure as heck don't understand how to set it up right now... although, I think the last time I tried to set it up, I was using v1 controllers... That may have been my issue, Either way, if a new HidGuardian comes out that fixes the issues, we won't have to rely on Exclusive mode anymore and can finally get rid of it. But until then, we are relying on exclusive mode, which is more of a bandage fix than a solution.

Also, thanks for merging the feature back in @Ryochan7 . I honestly wasn't expecting you to accept the pull request and was thinking I'd have to keep updating the fork. Which would have been a total nightmare! That code is WAY over my head!

**Comment by jedinjapan** [comment]
I understand the desire to depreciate this feature. However I find it very useful when using applications that support multiple input devices besides Xinput. DS4Windows is my main middleware for these controllers, so that whatever I launch just *works*. So be it Bioshock via Steam, Retroarch, or an old DOS title, I can pass the inputs that the title expects without it having native support. I generally run 1 profile for everything XInput, disable selectively if it has DS4 support natively, and a bit of custom remapping for 1 or 2 titles.

In the case of apps like Retroarch however, hiding the DS4 controller prevents it from incorrectly mapping the Wireless Controller 1 to primary control, and the XInput pad to P2. This is one of many examples.

I am willing to look into HIDGuardian going forward if @Ryochan7 feels this is the only realistic path, but wanted to add my voice that I appreciate you merging the feature back for now, and hopefully there is a viable way to keep it implemented without increasing the complexity of the code for you beyond something easily maintainable. Generally speaking, the feature works as intended for me on a clean boot, and can be "reset" if it is not working by restarting explorer. But if forced to choose between this and features like Dualsense haptics or triggers, if this is holding those back either in development or simply legacy support, then perhaps it is worth letting go.

**Comment by Aida-Enna** [comment]
Perhaps we could make a list of verified programs that interfere with HideDS4Controller that you could link people to in the error message? I think that would alleviate a lot of stress on the dev team, which is understandably an issue.

**Comment by mika-n** [comment]
This is why "Hide DS4 Controller" option and the issues with it is a problematic. Reasons are beyond the control of DS4Windows app, but even when there already is a documentation page (and ds4windows logs this URL in a logfile when there are exclusive mode error) then people don't read it.
https://github.com/Ryochan7/DS4Windows/wiki/Exclusive-Mode-(Hide-DS4-Controller-config-option)-tips-and-issues

**Comment by Aida-Enna** [comment]
Ah, I see. I didn't see the URL at the end of the error message (it's very long!), so it appears I now have egg on my face. 🙃

**Comment by jedinjapan** [comment]
Regardless of the above, HIDGuardian introduces significant complexity risks up to and including tampering with Mouse and KB level drivers to achieve this, and that seems well beyond the scope of what the average user could be asked to do in most cases. While "Hide DS4 Controller" still requires exclusive access, I think rebooting, or changing application priorities on boot is far more attainable for most than installing a HID Driver layer to permanently block the controller.

The HID Guardian solution also feels less than ideal for the limited number of PC games that *do* offer native dualshock support, as these titles will never be able to use it and always require DS4Windows even if strictly unnecessary due to proper support.

The current solution, while imperfect, is far more accessible and somewhat more easily resolved, in my opinion.

**Comment by RyuMakkuro** [comment]
> but even when there already is a documentation page (and ds4windows logs this URL in a logfile when there are exclusive mode error) then people don't read it.
> https://github.com/Ryochan7/DS4Windows/wiki/Exclusive-Mode-(Hide-DS4-Controller-config-option)-tips-and-issues

It would greatly help if people could actually see this link in the log. I've been using DS4Windows for many years now, but today I found out you can actually double click in the log on the error to have the whole error message pop up. So expecting people to read something when to even see the link you have to do very specific things that aren't super obvious is fairly naive. And that's ignoring the fact you have to manually type the whole link into the browser since you can't copy paste or simply click on it. Most people wouldn't have the patience for that. Heck, I myself would sooner google part of that link than to type it manually cause it would end up being faster.

I'd suggest putting that link (in a hyperlink format preferably or at least able to copy it) in the warning message that pops up when you enable Exclusive mode. Furthermore few extra tips could be added there, as in when the gamepad doesn't go into exclusive mode when it did before. Something like "stop and resume the service" or "re-connect the gamepad" or just straight up toggle the Exclusive mode off and on. That last one actually worked for me today when DualSense didn't want to get into exclusive mode whereas everything else that usually works didn't. Including those few simple troubleshooting steps should help a bit.

TL;DR; Don't hide the info people need to see behind logs/logfiles that don't immediately show entire message in the viewer, make it straight in their face the moment they even think about using it and super easy to access.

**Comment by MonfGeiger** [comment]
So out of everything here, this is what I got: Hide DS4 Controller option still in DS4Windows but legacy feature, no support for it at all so if shit pear shapes with it, you're S.O.L. Ok, now if I wanted to hide my ds4 controller if/when that legacy feature is erased once and for all, I need to use either HidGuardian or HidNinja? And there's a gui based program that can effectively add my ds4 (v1) controller to this list so if I run ds4windows on startup (task mode), it wont load 2 controllers (actual ds4 and virtual 360) but just the single virtual 360 controller.

Am I on the dot so far, or am I not getting anything? If so, please educate me, because I've been using ds4windows ever since....I wanna say scptoolkit ceased becoming an option

**Comment by Ryusennin** [comment]
HidNinja is the GUI for HidGuardian.

**Comment by mika-n** [comment]
Explanation of "double controller" issue and what it may cause in some games. And how to solve it via HideDS4Controller option or via HidGuardian driver tool (alternative solutions, not meant to be used both at the same time).
https://github.com/Ryochan7/DS4Windows/wiki/Exclusive-Mode-(Hide-DS4-Controller-config-option)-tips-and-issues

"Hide DS4 Controller" (=exclusive mode) works without any additional tools, but is it very fragile and may not always work because some other app has locked the device before DS4Windows app can do its "exclusive" magic. This is explained in the above linked web page. Those "other apps" are really beyond the control of DS4Windows. You have to try to figure out which app has locked the physical gamepad before DS4Windows can do the exclusive magic.

Or alternative solution is to use HidGuardian device driver to hide the physical gamepad. This works in 99.9999999% cases because the HidG kernel device driver can do so much more than a user space application (=DS4Windows and exclusive mode) can do.

HidNinja is then just an user interface to configure HidGuardian settings, so you don't have to do the HidG configuration manually via Windows registry editor. But, HidNinja on its own doesn't do anything and once you have used it to configure HidGuardian settings you can keep HidNinja closed (ie. no need to run HidNinja all the time in the background).

**Comment by MonfGeiger** [comment]
That clears it all up for me, thanks. ill make sure to copy/paste all that in a txt file for future reference

**Comment by tamodolo** [comment]
So I got here after formating my PC and studing a better than and less complex way to grant exclusive mode to DS4.

The main problem with the today's hidguardian is that it's a plague on the system. For some reason it's very sensitive and will break things if removed/unistalled (don't ask me why. I simply lost all my USBs after unstalling it las month)

HidHide is the new solution vigem group is working on. The ETA was end of 2020 so maybe we will see news about it.

Personaly I would like to avoid hidguardian as I would like to use some native game support more. This is mainly because vibration is inacurate on Vigem and don't really get any atention. Just read that it's now on driver cache and I really don't know if this makes it acurate now.

I'll write a guide in the wiki for this when a definitive solution rise.

**Comment by tonebacas** [comment]
> 
> 
> HidGuardian is more reliable way to hide the original physical controller, but the current version of it has issues with DS4v1 gamepads when a profile uses dualshock4 mode. DS4v2 gamepads work OK even in ds4 mode.

Would it be possible to emulate a virtual DS4v2 gamepad?

When you use HidGuardian to manage visibility for gamepads (physical or virtual), there seems to be an issue regarding DS4v1 controllers -- when I use DS4Windows to get my DS4v1 emulated to a virtual DS4 controller, and then use HidGuardian to give exclusive access to DS4Windows, the controller isn't detected ingame.

I read one user comment on how this is because HidGuardian can't tell the difference between the physical DS4v1 and the virtual controller, which I suppose would be a DS4v1 as well. For users who have a DS4v2, having their physical controller hidden from other processes besides DS4Windows causes no issue because the virtual DS4v1 controller is not being hidden.

If we had the ability to choose which DS4 version we would like to use for the virtual controller, we could then choose accordingly, so that HidGuardian doesn't hide both the physical and virtual devices (users of physical DS4v1 pick virtual DS4v2, hiding the physical controller).

**Comment by mika-n** [comment]
I have tried this with ViGem, but it didn't work. Didn't try to figure out too hard why it didn't work because the next version of HidGuardian presumably solves this anyway (it can be used to hide gamepads per instance and not per device type)

**Comment by Yohoki** [comment]
> 
> 
> > HidGuardian is more reliable way to hide the original physical controller, but the current version of it has issues with DS4v1 gamepads when a profile uses dualshock4 mode. DS4v2 gamepads work OK even in ds4 mode.
> 
> Would it be possible to emulate a virtual DS4v2 gamepad?
> 
> When you use HidGuardian to manage visibility for gamepads (physical or virtual), there seems to be an issue regarding DS4v1 controllers -- when I use DS4Windows to get my DS4v1 emulated to a virtual DS4 controller, and then use HidGuardian to give exclusive access to DS4Windows, the controller isn't detected ingame.
> 
> I read one user comment on how this is because HidGuardian can't tell the difference between the physical DS4v1 and the virtual controller, which I suppose would be a DS4v1 as well. For users who have a DS4v2, having their physical controller hidden from other processes besides DS4Windows causes no issue because the virtual DS4v1 controller is not being hidden.
> 
> If we had the ability to choose which DS4 version we would like to use for the virtual controller, we could then choose accordingly, so that HidGuardian doesn't hide both the physical and virtual devices (users of physical DS4v1 pick virtual DS4v2, hiding the physical controller).
There's a couple issues with this.

Firstly, emulating a v2 controller could be done, if the ViGEmBus drive was updated to allow it, and then that new functionality was implemented into D4W. But, that's a ViGEmBus issue, not D4W. Either way, that now makes it more complex and people already don't read about the exclusive mode issues. Not to mention, now the issue is flipped so that people with v2 controllers are getting double inputs.

Secondly, HidGuardian could be edited to hide v2 controllers instead of v1. Mika-n had made some progress there, but a new version is coming out soon(tm) that will address this issue anyway. But this again just flips the issue to v2 users having issues instead of v1 users.

Thirdly, what if you connect a v1 AND a v2 to play multiplayer? No matter what, there's issues. The only (current) solution is exclusive mode, which is a pain in the butt and doesn't work well. The better option would be the new version of HidG, whenever that comes out.

Hopefully, there's enough interest in the new HidG that a good tutorial gets made to show us how to set it up correctly. I tried using HidNinja, but didn't understand what I was doing... but I also had a v1 and v2 that I was trying to setup, which was probably part of the issue.

**Comment by Ryusennin** [comment]
I'm using both v1 and v2 controllers with HidGuardian. X360 mode is just fine. I only encounter problems if I ever need to use DS4 mode (with PS Remote Play for example). DS4 mode indeed doesn't work with a v1 pad, but that's the problem of the current ViGEm driver.

**Comment by Yohoki** [comment]
I also use both versions. I use the ds4 mode pretty often though. There's lots of games that will show PS prompts when using the ds4 mode, and even more games that just work but show basic or xbox controls. I've never use an xbox controller, so having xbox prompts almost entirely ruins the gameplay experience for me.... I can *usually* wrap my head around pressing the right button, but every once in a while you'll get a "Press X" and I don't know what to do for a second.... So, having DS4 mode working good is pretty important to me. It's worth it to get exclusive mode working right.

**Comment by tonebacas** [comment]
> people like me do exist, don't remove useful functionality in favor of people who break things by ticking boxes they don't understand

DS4Windows uses ViGem to do the controller emulation, and the people working on that are coming up with a different solution to effectively hide the controllers. At that point, DS4Windows doesn't need any controller hiding function -- you'll be using HidHide to give access of your physical controller to whichever applications you want. In this particular use, you'd want to hide your physical controller from all processes, except DS4Windows, granting it exclusive access.

This is what I understand from the replies in this issue, and from looking around in ViGem's page.

**Comment by Yohoki** [comment]
> 
> 
> just posting to say that I've been using an older build of DS4W because i find it works better for me than newer builds that made all these changes to things like this
> i have a v2 controller and have hotkeys set up to switch between outputting DS4 and outputting xinput with exclusive mode always enabled. switching profiles is quick and easy and there's never any duplicate controllers left behind "connected" in joy.cpl
> 
> with newer builds this became much clunkier, switching profiles has a delay, controller slots get swapped around, xinput controller stays "connected" even when i'm not using it etc it just doesn't suit me at all, it was so bad i switched back a while ago and have been using 1.7.28 ever since
> 
> people like me do exist, don't remove useful functionality in favour of people who break things by ticking boxes they don't understand
lol. I find it hard to beleive that removing this option would affect you at all if you're still on 1.7.28 .  XD

But, I do agree that a useful (although broken) feature shouldn't be removed until a replacement is found.


> 
> 
> > people like me do exist, don't remove useful functionality in favor of people who break things by ticking boxes they don't understand
> 
> DS4Windows uses ViGem to do the controller emulation, and the people working on that are coming up with a different solution to effectively hide the controllers. At that point, DS4Windows doesn't need any controller hiding function -- you'll be using HidHide to give access of your physical controller to whichever applications you want. In this particular use, you'd want to hide your physical controller from all processes, except DS4Windows, granting it exclusive access.
> 
> This is what I understand from the replies in this issue, and from looking around in ViGem's page.

The problem at the moment is that ViGEm's new feature isn't out yet. and until it is, exclusive mode is the only viable option for people with v1 AND v2 controllers. There's nothing wrong with D4W's exclusive mode code. it works flawlessly.... the issue is other programs... like, why does my graphics card need exclusive access to my controller? If you can answer that one, I'd love to know. XD

So, since the code isn't broken, and since there's no other viable solution available yet, there's no reason to remove it. When ViGEm or HidG finally releases the mythical unicorn that we've been told about, I will gladly get rid of clunky exclusive mode. But I'm not sure when this will be coming... or if... It's supposed to come out this year, but it's also not published, so we can't really see how things are going, either. All we can do is wait. :P

**Comment by RyuMakkuro** [comment]
> why does my graphics card need exclusive access to my controller? If you can answer that one, I'd love to know. XD

Assuming we're talking about the nVidia hardware, that's due to their streaming feature, aka nVidia Shield. It can be controlled through 3rd party controllers, so whenever Shadowplay is turned on, that functionality is also in there. Probably it makes it easier on their end for some things but who knows. Maybe just lazy coding for all we know.

**Comment by Ryochan7** [comment]
This thread ceased to be useful a long time ago
